### PR TITLE
webgl: Support colorMask in RenderPass

### PR DESCRIPTION
**Pull request by felixpalmer**
For https://github.com/visgl/deck.gl/pull/8636

When a WebGLRenderPass is created it [immediately does a clear()](https://github.com/visgl/luma.gl/blob/master/modules/webgl/src/adapter/resources/webgl-render-pass.ts#L36). deck.gl was previously setting a `colorMask` using `withParametersWebGL`, but now with this removed there is no way to set the `colorMask` before the clear operation. Doing it in `Pass.getLayerParameters()` is too late, as by then the `clear` operation has happened

#### Change List
- Support colorMask in RenderPass

### CODE REVIEW

**Comment by Pessimistress** [comment]
Defer to @ibgreen - is the WebGPU name `writeMask` preferred?

File: modules/core/src/adapter/types/parameters.ts
Comment (donmccurdy) [reply modules/core/src/adapter/types/parameters.ts:185]: Hm, there's also the RenderPipeline parameter, `colorMask`, which is a bit mask:

https://github.com/visgl/luma.gl/blob/f59d1d45aaa8c96ced2ea218d748d33cf162a9fd/modules/core/src/adapter/types/parameters.ts#L159-L160

**Review by ibgreen — APPROVED** [review]
FWIW

I believe clear colors in WebGPU are set per target (attachment) not as part of the props. 

For now, I don't think I exposed that full API.

Why can't we set it before we call gl.clear?

**Comment by Pessimistress** [comment]
> Why can't we set it before we call gl.clear?

We are trying to avoid using `setParametersWebGL`.

**Comment by felixpalmer** [comment]
OK, so it seems this change is OK, but we should change it to be a bitmask to align with the WebGPU API before merging

**Pull request merged**
